src/Microsoft.Health.Fhir.CosmosDb.Initialization/Features/Storage/DataPlaneCollectionSetup.cs

```diff
@@ -0,0 +1,159 @@
+            _container = new Lazy<Container>(() => cosmosClientInitializer.CreateFhirContainer(
+                _client,
+                cosmosDataStoreConfiguration.DatabaseId,
+                collectionId));
+
+            _logger = logger;
+        }
+
+        public async Task CreateDatabaseAsync(AsyncPolicy retryPolicy, CancellationToken cancellationToken = default)
+        {
+            EnsureArg.IsNotNull(retryPolicy, nameof(retryPolicy));
+
+            try
+            {
+                _logger.LogInformation("Creating Cosmos DB Database {DatabaseId}", _cosmosDataStoreConfiguration.DatabaseId);
+
+                if (_cosmosDataStoreConfiguration.AllowDatabaseCreation)
+                {
+                    _logger.LogInformation("CreateDatabaseIfNotExists {DatabaseId}", _cosmosDataStoreConfiguration.DatabaseId);
+
+                    await retryPolicy.ExecuteAsync(
+                        async () =>
+                            await _client.CreateDatabaseIfNotExistsAsync(
+                                _cosmosDataStoreConfiguration.DatabaseId,
+                                _cosmosDataStoreConfiguration.InitialDatabaseThroughput.HasValue ? ThroughputProperties.CreateManualThroughput(_cosmosDataStoreConfiguration.InitialDatabaseThroughput.Value) : null,
+                                cancellationToken: cancellationToken));
+                }
+
+                _logger.LogInformation("Cosmos DB Database {DatabaseId} successfully initialized", _cosmosDataStoreConfiguration.DatabaseId);
+            }
+            catch (Exception ex)
+            {
+                LogLevel logLevel = ex is RequestRateExceededException ? LogLevel.Warning : LogLevel.Critical;
+                _logger.Log(logLevel, ex, "Cosmos DB Database {DatabaseId} initialization failed", _cosmosDataStoreConfiguration.DatabaseId);
+                throw;
+            }
+        }
+
+        public async Task CreateCollectionAsync(IEnumerable<ICollectionInitializer> collectionInitializers, AsyncPolicy retryPolicy, CancellationToken cancellationToken = default)
+        {
+            EnsureArg.IsNotNull(_client, nameof(_client));
+            EnsureArg.IsNotNull(_cosmosDataStoreConfiguration, nameof(_cosmosDataStoreConfiguration));
+            EnsureArg.IsNotNull(collectionInitializers, nameof(collectionInitializers));
+            EnsureArg.IsNotNull(retryPolicy, nameof(retryPolicy));
+
+            try
+            {
+                _logger.LogInformation("Initializing Cosmos DB Database {DatabaseId} and collections", _cosmosDataStoreConfiguration.DatabaseId);
+
+                foreach (var collectionInitializer in collectionInitializers)
+                {
+                    await collectionInitializer.InitializeCollectionAsync(_client, retryPolicy, cancellationToken);
+                }
+
+                _logger.LogInformation("Collections successfully initialized");
```

Comment:
As I've mentioned in another comment, I believe we should install the stored procedures here. It'll keep the parity with the ARM Collection Setup.

Reply:
done